tests/unit/s2n_security_policy_cert_preferences_test.c

```diff
@@ -313,7 +313,11 @@ int main(int argc, char **argv)
      */
     {
         DEFER_CLEANUP(struct s2n_cert_chain_and_key *cert = NULL, s2n_cert_chain_and_key_ptr_free);
-        /* use a very insecure cert that would not be included in any reasonable cert preferences */
+        /* use a very insecure cert that would not be included in any reasonable cert preferences 
+         * 
+         * OpenSSL FIPS module requires the key size for RSA to be bigger than or equal to 1024 bits.
+         * See https://github.com/aws/s2n-tls/issues/4651 for more information.
+         */
```

Comment:
I don't think this is the right place for this comment. Without the context of the issue, nothing in the code really suggests we'd use <1024 bits here.

Although you could branch on libcrypto / FIPS here and run the test with 512 where possible. That would make this comment make sense.

Otherwise, it's tricky to say where the comment should live. This is useful information, so it really probably should live somewhere. Maybe on [the testlib methods](https://github.com/aws/s2n-tls/blob/b20075ba20f911e4c78bc3d2b79a35682763dbb6/tests/testlib/s2n_testlib.h#L212-L227?)?

And other than a comment, you could make a code change to make this clearer. You could check for r < 0 before you check for r == out->size in the rsa encrypt method. Then you could give it a more helpful error. I think the reason James was confused by the error was that it looked like the encryption was succeeding but with the wrong size because the error was "S2N_ERR_SIZE_MISMATCH".

Reply:
Thank you for the suggestion. I made a code change to catch this failure with a new err message, and also branched on openssl-fips to test this new error gets thrown correctly.